### PR TITLE
Chore: Copy logging and tenant code from Cortex

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
After discussing with @pstibrany, we've decided the best way to eliminate Loki's dependency on cortex/pkg/util/log and cortex/pkg/tenant is to copy them into Loki, since we think they're too trivial to fit dskit's topic.

The PR just copies cortex/pkg/util/log to pkg/logutil (named this way to avoid clashing with `go-kit/kit/log`) and parts of  cortex/pkg/tenant to pkg/util/tenant.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Comment by sandeepsukhani** [comment]
Thank you so much @aknuds1 for all the efforts you have been putting into these PRs.

> The PR just copies cortex/pkg/util/log to pkg/logutil (named this way to avoid clashing with go-kit/kit/log)
I think keeping it under `loki/pkg/util/log` would have been better for 2 reasons:
1. Avoid updating usages in the code which would have resulted in a smaller PR.
2. Avoid having yet another util package at the top.

What do you think, should we go with that? I don't mind doing that change if you want.

**Comment by aknuds1** [comment]
@sandeepsukhani I'm OK with that, but what do you think about my argument to not name the package `log` to avoid clashing with `go-kit/kit/log`? It's a lot of hassle to have to make an import alias, when making changes mechanically.

**Review by owen-d — APPROVED** [review]

**Comment by kavirajk** [comment]
Closing this as some of this worked moved into another PR #4944 . Thanks @aknuds1 :)

File: pkg/logutil/log.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/weaveworks/common/logging"
+	"github.com/weaveworks/common/server"
+)
+
+var (
+	// Logger is a shared go-kit logger.
+	// TODO: Change all components to take a non-global logger via their constructors.
+	// Prefer accepting a non-global logger as an argument.
+	Logger = log.NewNopLogger()
```
Comment (jeschkies) [inline pkg/logutil/log.go:19]: Thanks for you effort! I wonder whether we should rather get rid off this logger instead as the todo suggest. What do you think?

**Pull request closed without merging**